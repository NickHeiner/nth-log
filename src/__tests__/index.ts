import createLog, {constantizeLogEntryForTest} from '..';
import {RingBuffer} from 'bunyan';

const wait = (timeMs: number) => new Promise(resolve => setTimeout(resolve, timeMs));

// I'm ok with magic numbers in a test.
/* eslint-disable no-magic-numbers */

it('logger', async () => {
  const ringBuffer = new RingBuffer({ limit: 100 });
  const logger = createLog({
    name: 'test-log', 
    streams: [{
      level: 'trace',
      type: 'raw',
      stream: ringBuffer
    }],
    stream: undefined
  });
  logger.trace({dataEntry: 'data value trace'}, 'String label');
  logger.debug({dataEntry: 'data value debug'}, 'String label');
  logger.info({dataEntry: 'data value info'}, 'String label');
  logger.warn({dataEntry: 'data value warn'}, 'String label');
  logger.error({dataEntry: 'data value error'}, 'String label');
  logger.fatal({dataEntry: 'data value fatal'}, 'String label');

  await logger.logPhase({phase: 'phase name'}, async (logProgress, setAdditionalLogMetadata) => {
    await wait(10);
    logProgress({substep: 1}, 'First wait complete');
    await wait(10);
    logProgress({substep: 2}, 'Second wait complete');
    setAdditionalLogMetadata({additional: 'metadata'});
  });

  expect(ringBuffer.records.map(record => constantizeLogEntryForTest(record))).toMatchSnapshot();
});