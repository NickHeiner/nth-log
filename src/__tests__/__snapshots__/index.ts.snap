// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger 1`] = `
Array [
  Object {
    "dataEntry": "data value trace",
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 10,
    "msg": "String label",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "dataEntry": "data value debug",
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 20,
    "msg": "String label",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "dataEntry": "data value info",
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 30,
    "msg": "String label",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "dataEntry": "data value warn",
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 40,
    "msg": "String label",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "dataEntry": "data value error",
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 50,
    "msg": "String label",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "dataEntry": "data value fatal",
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 60,
    "msg": "String label",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 30,
    "msg": "Starting phase name",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 30,
    "msg": "In progress: phase name",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "substep": 1,
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 30,
    "msg": "In progress: phase name",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "substep": 2,
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
  Object {
    "additional": "metadata",
    "durationMs": 117,
    "hostname": "<hostname placeholder>",
    "level": 30,
    "msg": "Completed phase name",
    "name": "test-log",
    "pid": 9001,
    "prettyDuration": "<duration placeholder>",
    "time": 2012-01-01T00:00:00.000Z,
    "v": 0,
  },
]
`;
